Fixes to get rpi mission working with recent NMF

sachavg opened the pull request:
This PR includes four small fixes to the code to get it to compile and upload to an RPi, including updated documentation to reflect changes. 

Another issue seemed to be better to fix at the framework level, that is described in the tiny pull request at: https://github.com/esa/nanosat-mo-framework/pull/184 

Let me know if you want me to split this up, or alter in any way, 

thanks!

CesarCoelho commented:
Hi Sacha,

Thanks for your contribution!
Is this working with the phi-sat-2 branch? Eventually the phi-sat-2 will become the main branch, so I don't want to go back and then to upgrade it again to phi-sat-2. That would be double effort.

Best regards,
César Coelho

sachavg commented:
Yes, it actually only works with the phi-sat-2 branch, as the code already expects some dependencies from it.
(it however only works if https://github.com/esa/nanosat-mo-framework/pull/184 is merged to the phi-sat-2 branch)

CesarCoelho commented:
I think you compile it against the dev branch because the phi-sat-2 has the new AI service that you removed.
Also, I removed the orekit dependency that was just bloating the deployment, I see that you are adding it back.
Try deleting your .m2 repository and compiling from scratch with the phi-sat-2 branch.  :)

sachavg commented:
I added the updated readme and will check a clean install to see if i indeed mixed the branches up. I was playing around with the framework before, so that is very much possible. I will alter this PR accordingly. 

`ArtificialIntelligenceInheritanceSkeleton()` was not removed, but it was also defined at the bottom of the same file, so the compiler did not like that, and i removed one of the instances. (not sure if I removed the right one though)

sachavg commented:
I just checked everything again, and compiled against phi-sat-2
Without `orekit` I get the following error regarding `GPSSoftSimAdapter`: 

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.3:compile (default-compile) on project nmf-platform-impl-raspberrypi: Compilation failure: Compilation failure: 
[ERROR] /home/sacha/os3/nmf-mission-raspberry-pi/platform-service-impl/src/main/java/esa/mo/platform/impl/provider/raspberrypi/GPSSoftSimAdapter.java:[30,45] package org.orekit.propagation.analytical.tle does not exist
[ERROR] /home/sacha/os3/nmf-mission-raspberry-pi/platform-service-impl/src/main/java/esa/mo/platform/impl/provider/raspberrypi/GPSSoftSimAdapter.java:[63,12] cannot find symbol
[ERROR]   symbol:   class TLE
[ERROR]   location: class esa.mo.platform.impl.provider.raspberrypi.GPSSoftSimAdapter
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

Adding orekit solves this error

sachavg commented:
Hi @CesarCoelho sorry for the commit mess, but I have now removed orekit from the code, so the dependency could be removed again, and further fixed the docker and rp environment so it works  with the phi-sat-2 branch. Except for the PR still open in that branch, let me know if you rather have me add that here, although I think my proposed change in https://github.com/esa/nanosat-mo-framework/pull/184 makes sense to implement never-the-less.

CesarCoelho commented:
I am making this repo work with the latest version 4.0 of the NMF.
Thank you so much for your contribution!
